opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
+        # Calculate a0
+        a0 = Emax * const.e / (const.m_e * const.c * omega)
+        return( a0 )
+
+    def get_ctau( self, t=None, iteration=None, pol=None ):
+        """
+        Calculate the length of a (gaussian) laser pulse. ( sqrt(2) * sigma_z)
+
+        Parameters
+        ----------
+        t : float (in seconds), optional
+            Time at which to obtain the data (if this does not correspond to
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        pol : float
+            Polarization angel of the field relative to the x plane. Can be
+            freely chosen between 0 and Pi/2 in thetaMode. For carthesian
+            coordinates it as to be either 0 or Pi/2 (x or y plane)
+
+        Returns
+        -------
+        Float with ctau in meters
+        """
+        # Get the field envelope
+        E, extent = self.get_laser_envelope(t=t, iteration=iteration,
+                                            pol=pol)
+        # Calculate standard deviation
+        sigma = wstd(extent.z, E)
+        # Return ctau = sqrt(2) * sigma
+        return( np.sqrt(2) * sigma )
+
+    def get_laser_waist( self, t=None, iteration=None, pol=None ):
+        """
+        Calculate the waist of a (gaussian) laser pulse. ( sqrt(2) * sigma_r)
+
+        Parameters
+        ----------
+        t : float (in seconds), optional
+            Time at which to obtain the data (if this does not correspond to
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        pol : float
+            Polarization angel of the field relative to the x plane. Can be
+            freely chosen between 0 and Pi/2 in thetaMode. For carthesian
+            coordinates it as to be either 0 or Pi/2 (x or y plane)
+
+        Returns
+        -------
+        Float with laser waist in meters
+        """
+        # Get the field envelope
+        field, extent = self.get_laser_envelope(t=t, iteration=iteration,
+                                                pol=pol, index='all')
+
+        # Find the maximum of the envelope along the transverse axis
+        trans_max = np.amax(field, axis=1)
+        # Get transverse positons
+        trans_pos = getattr(extent, extent.axes[0])
+        # Calculate standard deviation
+        sigma_r = wstd(trans_pos, trans_max)
+        # Return the laser waist = sqrt(2) * sigma_r
+        return(np.sqrt(2) * sigma_r)
+
+    def wigner_transform( self, t=None, iteration=None, pol=None ):
```

Comment:
Nice ! Another really great feature !!!

I have a suggestion on the API: why not return a (properly initialized) `FieldMetaInformation` object along with the wigner transform. That way, the user can plot the wigner transform by using
`plt.imshow( wigner, extent=info.imshow_extent )`

See below for how the FieldMetaInformation could be initialized.